packages/server/src/middlewares/routes.ts

```diff
@@ -1,10 +1,21 @@
 import Router from '@koa/router'
+import multer from '@koa/multer'
+import { DefaultState, DefaultContext } from 'koa'
 
 import { FilesController } from '../controllers'
 
-const router = new Router()
+const MEGABYTES = 10 ** 6
+const upload = multer({
+  limits: {
+    fileSize: 10 * MEGABYTES
+  },
+  dest: '/uploads'
+})
+
+const router = new Router<DefaultState, DefaultContext>()
+
 router.get('/', (ctx) => ctx.body = { up: true })
-router.post('/files', FilesController.create)
+router.post('/files', upload.single('file'), FilesController.create)
```

Comment:
I'd try to handle the upload in the `create` function itself. Additionally, we can control the file size with the validator to avoid handling a custom error in the top-level error handler giving a custom message about what failed to the user.

Reply:
I thought about this, but:
1) multer is the most battle tested file upload middleware in express and koa, not worth rewriting.
2) multer handles file size limits while the request is streaming. If I want to test size in the validator I have to upload the whole file to disk whatever the size so I can check later. Or I stream in the validator and throw error when size exceeds, but then it's much more than a validator at that point.

So I decided to have it like this and just handle a special error in the error handler, not a big deal (we can also do custom messages there if we wish)

Reply:
Fair, what about simply moving the upload logic to the controller then?

Reply:
Not sure what you mean, `upload.single` uploads the file, adds middleware parameters and then I handle it in the next one. How would you move everything to controller?

Reply:
I was thinking kind of mixing the two middlewares in one but maybe it looks weird, at least I'd encapsulate it in another function of the `FilesController` so it can look sth like:

```js
router.post('/files', FilesController.upload, FilesController.create)
```

it could be done as follows:

```js
const MEGABYTES = 10 ** 6
const MAX_FILE_SIZE = 10 * MEGABYTES

export default class FilesController {
  ...
  static upload: Middleware = async (ctx) => {
    const upload = multer({ limits: { fileSize: MAX_FILE_SIZE }, dest: '/uploads' })
    return upload.single('file')
  }
}
```

Reply:
I was thinking kind of mixing the two middlewares in one but maybe it looks weird, at least I'd encapsulate it in another function of the `FilesController` so it can look sth like:

```js
router.post('/files', FilesController.upload, FilesController.create)
```

it could be done as follows:

```js
const MEGABYTES = 10 ** 6
const MAX_FILE_SIZE = 10 * MEGABYTES

export default class FilesController {
  ...
  static upload: Middleware = async (ctx) => {
    const upload = multer({ limits: { fileSize: MAX_FILE_SIZE }, dest: '/uploads' })
    return upload.single('file')
  }
}
```

Reply:
gotcha, yes I like this better

Reply:
added `static upload: Middleware = upload.single('file')`